### DISCUSSION TITLE
Bedingte Suche bei abhängigen Suchfeldern

### DISCUSSION THREAD

**Discussion by nicole-vlbg** [e-netze/webgis-community#141] [Q&A]
Wir hatten eine Anwenderanfrage bzgl. der "Abhängigen Suche" in der Detailsuche. Der Suchalgorithmus lässt dies aber derzeit nicht zu. Und zwar geht es um Folgendes:

Suche von Wanderrouten
Im ersten Suchfeld wird der Typ ausgesucht
![image](https://github.com/gis-eni/webgis-forum/assets/124671123/2169d76f-a40a-42c3-a9be-851a1bca5204)
 
Im zweiten erscheint dann entsprechend der ersten Auswahl eine Dropdownliste mit den Routennamen 
![image](https://github.com/gis-eni/webgis-forum/assets/124671123/b5aad582-0b09-4a42-acd6-cc948acad0bb)

Jetzt soll es aber noch eine Möglichkeit geben die quasi alle Routennamen unabhängig vom Typ anzeigt, d.h. beim Suchfeld  Typ würde dann nichts ausgewählt oder eine leeres Feld.

Wäre es machbar, dass der Suchalgorithmus eine bedingte Suche in dieser Form berücksichtigt?

IF 
ART NOT IN ('Hauptroute','lokale Route','Weitwanderweg')
UPPER(ROUTENNAME) LIKE UPPER('[routenname]%')
ELSE
ART = '[typ]' AND UPPER(ROUTENNAME) LIKE UPPER('[routenname]%')

**Comment by gis-eni** [comment]
So ganz blauäugig würde ich sagen:

Select ROUTENAME from ... where ART like '[typ]%'

=> wenn nichts eingegeben wurde, dann sollte ART like '%' alles zurück liefern, oder?

**Comment by gis-eni** [comment]
2. Variante:
Bau zwei Abfragen
* Wanderroute (nach Typ/Name) - entspricht der, die du schon hast
* Wandroute (nach Name) - Hier lässt du den Typ weg und der Anwender kann nur den Namen eingeben (mit einfacher Auswahlliste)
